### ISSUE TITLE
Get color from image if the querySelector is an image element

### ISSUE THREAD

**Issue by Explosion-Scratch** [easonwong-de/Adaptive-Tab-Bar-Colour#79]
In this function:
https://github.com/easonwong-de/Adaptive-Tab-Bar-Colour/blob/909550903145f67ff34b3ac95557ac6c2a991d41/content_script.js#L330-L334

**Comment by Explosion-Scratch** [comment]
I created a Userscript which does this, but it would be nice to see it in the extension as well as an option:
https://gist.github.com/Explosion-Scratch/bb7e6e9db855cd51bb860e132ee3cb05

**Comment by Explosion-Scratch** [comment]
<img width="1362" alt="image" src="https://github.com/user-attachments/assets/b06d0eff-55ad-44b5-af75-ac4e0e3075d8">
<img width="1300" alt="Screen Shot 2024-07-25 at 10 30 45 AM" src="https://github.com/user-attachments/assets/e959be95-5b94-40d1-9531-416676d2e927">

It looks beautiful 🩷

**Comment by easonwong-de** [comment]
This is absolutely cool! Thanks for the suggestion!

**Comment by easonwong-de** [comment]
I’m thinking about adding a “use userscript” option in the site list page, for example:

![Screenshot 2024-12-06 at 10-16-53 Add-ons Manager](https://github.com/user-attachments/assets/cfbbb905-f030-4c1e-9052-e96e79b1f6da)

**Comment by Explosion-Scratch** [comment]
What, how would that work, would you use a custom userscript engine? Just use an image as the querySelector, that would fix everything. Then more work could be done optionally with a userscript if the user watns

**Comment by easonwong-de** [comment]
Hi, I just found out the effect has always been quite easy to achieve, I just don’t know if this was how you got the screenshots:

With the said userscript installed, I added a rule for `www.spotify.com`: select “pick colour for element”, and fill the textbox with `#current_album_accent`.

**Comment by easonwong-de** [comment]
My settings look like this:

![image](https://github.com/user-attachments/assets/188352dd-aaa1-44a8-bd34-6de210c623ed)

And it works pretty well.

![image](https://github.com/user-attachments/assets/4335d389-e141-4f58-a3be-6fc811185375)

I am presuming, you prefer the add-on to fetch average / accent colour off images directly. Well, since the ATBC + Userscript bundle works so well, I don't see the necessity to implement it.

One thing I think I will improve is, the add-on will detect the change of a HTML element, if the element is set to be the source of ATBC colour (for instance, in this case, `#current_album_accent`).

**Comment by Explosion-Scratch** [comment]
@easonwong-de Yeah, that's because I created the userscript. The userscript wouldn't be necessary if this extension could just get the color from the image. My userscript gets the color from an image and converts it to an element, but is specific to spotify. That's why I want it to be added to the extension

**Comment by easonwong-de** [comment]
Thanks for the clarification! Now I get it: `getComputedStyle(element).backgroundColor` doesn’t work for a lot of elements, for example images. The colour picking method can be expanded to support more websites & custom rules.

**Comment by Explosion-Scratch** [comment]
> Thanks for the clarification! Now I get it: `getComputedStyle(element).backgroundColor` doesn’t work for a lot of elements, for example images. The colour picking method can be expanded to support more websites & custom rules.

Just add image color detection support
